src/app/editor-tabs/ace.component.ts

```diff
@@ -21,21 +23,38 @@ declare var createXtextEditor: (config: any) => Deferred;
   templateUrl: './ace.component.html',
   styleUrls: ['./ace.component.css']
 })
-export class AceComponent implements AfterViewInit {
+export class AceComponent implements AfterViewInit, OnDestroy {
 
   static readonly UNKNOWN_LANGUAGE_SYNTAX_PATH = 'none';
 
   @Input() path: string;
   @Input() tabId: string;
   editor: Promise<any>;
 
+  subscriptions: Subscription[] = [];
```

Comment:
a simple field (instead of an array) would also work, if additional subscriptions are added to the initial one as children (with `subscription.add`, I think). This has the benefit that all subscriptions will automatically be cancelled when `unsubscribe` is called on the root. See http://brianflove.com/2016/12/11/anguar-2-unsubscribe-observables/ (unfortunately, we cannot use the more elegant idioms described there, because our message bus returns a subscription instead of an observable…)